src/__tests__/clowder.test.ts

```diff
@@ -212,27 +242,30 @@ describe('Clowder Class', () => {
     expect(nextItem?.id).toMatch(/^(0|1)$/);
   });
 
-  it('should select an unvalidated item if no validated items remain', () => {
+  it('should return an item from missing if no validated items remain and returnUndefinedOnExhaustion is false', () => {
     const clowderInput: ClowderInput = {
       cats: {
         cat1: { method: 'MLE', theta: 0.5 },
+        cat2: { method: 'EAP', theta: -1.0 },
       },
       corpus: [
-        createMultiZetaStimulus('0', [createZetaCatMap([])]),
-        createMultiZetaStimulus('1', [createZetaCatMap(['cat1'])]),
-        createMultiZetaStimulus('2', [createZetaCatMap([])]),
+        createMultiZetaStimulus('0', [createZetaCatMap(['cat2'])]), // Validated for cat2
+        createMultiZetaStimulus('1', [createZetaCatMap(['cat2'])]), // Validated for cat2
+        createMultiZetaStimulus('2', [createZetaCatMap(['cat2'])]), // Validated for cat2
       ],
     };
+
     const clowder = new Clowder(clowderInput);
 
+    // Attempt to select an item for cat1, which has no validated items in the corpus
     const nextItem = clowder.updateCatAndGetNextItem({
       catToSelect: 'cat1',
-      catsToUpdate: ['cat1'],
-      items: [clowder.corpus[1]],
-      answers: [1],
+      returnUndefinedOnExhaustion: false, // Ensure fallback is enabled
     });
+
+    // Should return an item from `missing`, which are items validated for cat2
     expect(nextItem).toBeDefined();
-    expect(['0', '2']).toContain(nextItem?.id);
+    expect(['0', '1', '2']).toContain(nextItem?.id); // Item ID should match any of the items for cat2
```

Comment:
And then here, since we're expecting random behavior, can you do several iterations of this `expect`. Like

```suggestion
    const nDraws = 50;
    for (const _ of Array(nDraws).fill(0)) {
      expect(['0', '1', '2']).toContain(nextItem?.id); // Item ID should match any of the items for cat2
    }
```